src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,31 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { ReactElement, useState } from "react";
+import TreeNode, { TreeNodeProps } from "./Node";
+import css from "./Tree.module.css";
+
+export interface TreeProps {
+    nodes: TreeNodeProps[];
+    onSelect: (id: string) => void;
+    activeNodeId?: string;
+}
+
+export default function Tree({
+    nodes,
+    onSelect,
+    activeNodeId: initialActiveNodeId,
+}: TreeProps): ReactElement<TreeProps> {
+    const [activeNodeId, setActiveNodeId] = useState<string | undefined>(initialActiveNodeId);
```

Comment:
here you need the `useEffect(() => setActiveNodeId(initialActiveNodeId), [initialActiveNodeId]);` logic otherwise the component does not rerender if you pass it a new `activeNodeId` 😉

Reply:
thx, good catch: adaf607